### DISCUSSION TITLE
Recommendations for new project; wait for Play 2.9 or go with current 2.8 and Akka?

### DISCUSSION THREAD

**Discussion by Knitter** [playframework/playframework#11968] [Q&A] [answered]
Hi,

I'm going to start using Play to develop some integration services/apps and in one particular application I'll be using SSE.  Given the expected 2.9 released, Scala 3, Akka to be replaced (will it?), etc., would you recommend waiting for Play 2.9 or starting now with the latest version and upgrade/migrate later?

I'll be using Scala but I'm really new to it, and also kind in doubt about how easy it will be to move from Scala 2 to Scala 3.

I appreciate any suggestions you may have, though I understand this is less of a Q&A and more of an opinion based issue.

Thank you.

**Comment by Squeng** [comment, marked as answer]
I would wait for Play 2.9 and directly start with Scala 3.

In the meantime, since you're new to Scala (welcome!), you could gain more experience with Scala 3 by studying
- https://horstmann.com/scala/index.html or
- https://www.fcpbook.org/ (highly recommended, even for non-Scala programmers)
(And take advantage of https://www.informit.com/promotions/back-to-learning-sale-save-50-on-books-ebooks-and-video-142520)

I hope this helps!

PS: https://www.manning.com/books/functional-programming-in-scala-second-edition is also great so far, but I'm not done with it yet, which is why I haven't added it to [my list](https://www.squeng.com/stapel) yet.

**Reply by jducoeur** [reply]
> PS: https://www.manning.com/books/functional-programming-in-scala-second-edition is also great so far, but I'm not done with it yet, which is why I haven't added it to [my list](https://www.squeng.com/stapel) yet.

_Functional Programming in Scala_ is great, and well worth reading, but do *not* start there.  It isn't a book that teaches Scala, it's a book that teaches Functional Programming *using* Scala -- you should know the language at least somewhat before starting in on it.

(I've seen too many people start learning Scala with FPiS and wind up frustrated with both Scala and FP as a result, so I caution folks away from doing that.)

If you do read FPiS down the line, be sure to do all the exercises: much of the value of the book comes in the fact that it's really a *course* in FP, and most people learn it much better by actually getting their hands dirty and writing the code snippets.

Which book is best to start with depends on which language(s) you already know, but I agree that _Scala for the Impatient_ (linked above) is often a great place to start.

**Reply by mkurz** [reply]
Play 2.9 release candidate is now available, so no need to wait for anything anymore :wink: https://github.com/playframework/playframework/releases/tag/2.9.0-RC2

**Reply by Squeng** [reply]
Thanks a lot, @mkurz! I'm looking forward to giving it (as well as Java 21) a try soon.

**Reply by Knitter** [reply]
Thank you all for your answers, couldn't reply earlier, but I've already been looking for those books and will move my POC/demo to Play2.9 ASAP :)

**Reply by mkurz** [reply]
@Knitter Play 2.9 and 3.0 have been released now: https://twitter.com/playframework/status/1721576472047911007

**Reply by Knitter** [reply]
@mkurz yep, already looking at it. Going for Play 3.0. Thanks.
